### ISSUE TITLE
compile error ".parse" is not a member

### ISSUE THREAD

**Issue by HerrDingenz** [gabime/spdlog#2805]
After updating spdlog to the latest version I get the compile error: ".parse is not a member of ......"

My code worked perfectly fine before I updated spdlog. It seems like the commit [62e55e7](https://github.com/gabime/spdlog/commit/62e55e7a7fb2fd3fd43fced10cccf166fcd8ff40) broke it.

Here's an example of what isn't working anymore. I have a struct with some data in it that i want to log:

```
struct SomeData
{
   int a = 0;
   float b = 0.0f;
   bool c = false;

   friend std::ostream& operator << (std::ostream& os, const SomeData& data) noexcept
   {
      return os << std::boolalpha << "Data: a = " << data.a << ", b = " << data.b << ", c = " << data.c << '\n';
   }
};

int main()
{
   SomeData data{10, 23.23f, true};
   std::cout << data;				// works

   // let's assume I have a proper working spdlog::logger for the console that also includes <spdlog/fmt/ostr.h>
   my_loggger->trace(data);			// ERROR: .parse is not a member
}
```

If I construct the object and send it std::cout it works like expected and prints the data to the console. However when sending it to spdlog it won't compile.
When I checkout the commit right before the previously mentioned one ( [b85c509](https://github.com/gabime/spdlog/commit/b85c509ec670fb195f19cd63b914cd6f79311e5a) ) everything works fine.

EDIT: I forgot to mention I am in VS2022, c++latest (C++23)

**Comment by gabime** [comment]
See the [example](https://github.com/gabime/spdlog/blob/v1.x/README.md#user-defined-types) how to format your own classes in fmtlib 10.

**Comment by HerrDingenz** [comment]
Thanks for your response. So you're telling me to change my whole codebase and write formatters for every single type just because you upgraded to fmtlib 10? It all worked perfectly fine with my code example before you upgraded to fmtlib 10.

**Comment by gabime** [comment]
You don't have to upgrade. It will be released in spdlog version 2.0 since it is a breaking change.

**Comment by gabime** [comment]
As a quick fix, note you can just do
``` c++
#include "spdlog/fmt/ostr.h" // support for user defined types
template <> struct fmt::formatter<SomeData> : ostream_formatter {};
```

In any case, I think I will revert the upgrade to fmt 10 in this branch to avoid confusion (and re-apply in version 2 branch).

**Comment by gabime** [comment]
Fixed in 4338b9cd23c90c7bacde5bfefddade8d02dfd821

**Comment by HerrDingenz** [comment]
thank you .. now everything works again on latest commit without me having to write all those formatters myself :)
